src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionSecurityTokenAuthenticator.cs

```diff
@@ -0,0 +1,1366 @@
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("value");
+                }
+                this.issuerBindingContext = value.Clone();
+            }
+        }
+
+        public SecurityBindingElement BootstrapSecurityBindingElement
+        {
+            get { return this.bootstrapSecurityBindingElement; }
+            set
+            {
+                this.CommunicationObject.ThrowIfDisposedOrImmutable();
+                if (value == null)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("value");
+                }
+                this.bootstrapSecurityBindingElement = (SecurityBindingElement)value.Clone();
+            }
+        }
+
+        public IMessageFilterTable<EndpointAddress> EndpointFilterTable
+        {
+            get
+            {
+                return this.endpointFilterTable;
+            }
+            set
+            {
+                this.CommunicationObject.ThrowIfDisposedOrImmutable();
+                this.endpointFilterTable = value;
+            }
+        }
+
+        public Uri ListenUri
+        {
+            get { return this.listenUri; }
+            set
+            {
+                this.CommunicationObject.ThrowIfDisposedOrImmutable();
+                this.listenUri = value;
+            }
+        }
+
+        public virtual XmlDictionaryString IssueAction
+        {
+            get
+            {
+                return standardsManager.SecureConversationDriver.IssueAction;
+            }
+        }
+
+        public virtual XmlDictionaryString IssueResponseAction
+        {
+            get
+            {
+                return standardsManager.SecureConversationDriver.IssueResponseAction;
+            }
+        }
+
+        public bool PreserveBootstrapTokens
+        {
+            get
+            {
+                return this.preserveBootstrapTokens;
+            }
+            set
+            {
+                this.preserveBootstrapTokens = value;
+            }
+        }
+
+        public virtual XmlDictionaryString RenewAction
+        {
+            get
+            {
+                return standardsManager.SecureConversationDriver.RenewAction;
+            }
+        }
+
+        public virtual XmlDictionaryString RenewResponseAction
+        {
+            get
+            {
+                return standardsManager.SecureConversationDriver.RenewResponseAction;
+            }
+        }
+
+        public virtual XmlDictionaryString CloseAction
+        {
+            get
+            {
+                return standardsManager.SecureConversationDriver.CloseAction;
+            }
+        }
+
+        public virtual XmlDictionaryString CloseResponseAction
+        {
+            get
+            {
+                return standardsManager.SecureConversationDriver.CloseResponseAction;
+            }
+        }
+
+        //public bool RemoveCachedLogonToken(string username)
```

Comment:
This is used in the scenario where username/password auth is being used and the authentication mode is Windows auth. Basically we take the username/password provided and login to Windows. We then cache a token which contains all the claims info (groups the user belongs to etc) so we don't need to do an actual Login to achieve this. This is a problem for CoreWCF for a few reasons. First, it requires a P/Invoke calls which I'm avoiding. Second we would need to work out the equivalent on Linux. Third, it requires the client to send the username/password in the raw. That last issue I really don't like. We should be way past sending the password to the server. Anyway, we need to decide what we're going to support when it comes to username/password authentication. WCF only supports the Basic Username profile 1.0. 1.1 allows sending a hashed password instead. This won't allow using Windows Auth, but might be an option for Custom. Anyway, can you open an issue for deciding what to do about Username/Password authentication in general because if we're going to support it, we need this code working.

Reply:
https://github.com/CoreWCF/CoreWCF/issues/262